sklearn/partial_dependence.py

```diff
@@ -0,0 +1,590 @@
+          method, you can use a subset of ``X`` or a more coarse grid.
+        - If 'estimated', the function will be calculated by calling the
+          ``predict_proba`` method of ``est`` for classification or ``predict``
+          for regression on the mean of ``X``.
+        - If 'auto', then 'recursion' will be used if ``est`` is
+          BaseGradientBoosting or ForestRegressor, and 'exact' used for other
+          estimators.
+    n_jobs : int
+        The number of CPUs to use to compute the PDs. -1 means 'all CPUs'.
+        Defaults to 1.
+    verbose : int
+        Verbose output during PD computations. Defaults to 0.
+    ax : Matplotlib axis object, default None
+        An axis object onto which the plots will be drawn.
+    line_kw : dict
+        Dict with keywords passed to the ``matplotlib.pyplot.plot`` call.
+        For one-way partial dependence plots.
+    contour_kw : dict
+        Dict with keywords passed to the ``matplotlib.pyplot.plot`` call.
+        For two-way partial dependence plots.
+    fig_kw : dict
+        Dict with keywords passed to the figure() call.
+        Note that all keywords not recognized above will be automatically
+        included here.
+
+    Returns
+    -------
+    fig : figure
+        The Matplotlib Figure object.
+    axs : seq of Axis objects
+        A seq of Axis objects, one for each subplot.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import make_friedman1
+    >>> from sklearn.ensemble import GradientBoostingRegressor
+    >>> X, y = make_friedman1()
+    >>> clf = GradientBoostingRegressor(n_estimators=10).fit(X, y)
+    >>> fig, axs = plot_partial_dependence(clf, X, [0, (0, 1)]) #doctest: +SKIP
+    ...
+    """
+    import matplotlib.pyplot as plt
+    from matplotlib import transforms
+    from matplotlib.ticker import MaxNLocator
+    from matplotlib.ticker import ScalarFormatter
+    # TODO: The pattern below required to avoid a namespace collision.
+    # TODO: Move below imports to module level import at 0.22 release.
+    from .ensemble.gradient_boosting import BaseGradientBoosting
+    from .ensemble.forest import ForestRegressor
+
+    if method == 'auto':
+        if isinstance(est, (BaseGradientBoosting, ForestRegressor)):
+            method = 'recursion'
+        else:
+            method = 'exact'
+    if (not isinstance(est, (BaseGradientBoosting, ForestRegressor)) and
+            method == 'recursion'):
+        raise ValueError('est has to be an instance of BaseGradientBoosting or'
+                         ' ForestRegressor for the "recursion" method. Try '
+                         'using method="exact" or "estimated".')
+    if (not hasattr(est, '_estimator_type') or
+            est._estimator_type not in ('classifier', 'regressor')):
+        raise ValueError('est must be a fitted regressor or classifier model.')
+    if method != 'recursion' and est._estimator_type == 'classifier':
```

Comment:
`and not hasattr(est, 'predict_proba')` is missing, isn’t it?

Reply:
Yeah, the `plot_partial_dependence` function is lagging a bit behind the worker bee `partial_dependence` ... Shall bring up to speed now...